File: src/main/java/com/lmp/web/dto/RegisterDto.java
package com.lmp.web.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * DTO pour l'inscription d'un nouvel utilisateur.
 * Inscription simplifiée : seuls email et password sont obligatoires.
 * Le pseudo d'affichage sera automatiquement généré depuis l'email.
 */
public class RegisterDto {

    @Size(max = 50, message = "Le prénom ne peut pas dépasser 50 caractères")
    private String firstName;

    @Size(max = 50, message = "Le nom ne peut pas dépasser 50 caractères")
    private String lastName;

    @NotBlank(message = "L'email est obligatoire")
    @Email(message = "Format d'email invalide")
    @Size(max = 100, message = "L'email ne peut pas dépasser 100 caractères")
    private String email;

    @NotBlank(message = "Le mot de passe est obligatoire")
    @Size(min = 6, max = 100, message = "Le mot de passe doit contenir entre 6 et 100 caractères")
    private String password;

    @NotBlank(message = "La confirmation du mot de passe est obligatoire")
    private String confirmPassword;

    @Size(max = 20, message = "Le téléphone ne peut pas dépasser 20 caractères")
    private String phone;

    @Size(max = 255, message = "L'adresse ne peut pas dépasser 255 caractères")
    private String address;

    @Size(max = 50, message = "La ville ne peut pas dépasser 50 caractères")
    private String city;

    @Size(max = 20, message = "Le code postal ne peut pas dépasser 20 caractères")
    private String postalCode;

    @Size(max = 50, message = "Le pays ne peut pas dépasser 50 caractères")
    private String country;

    @Size(max = 100, message = "Le nom de l'entreprise ne peut pas dépasser 100 caractères")
    private String companyName;

    private boolean acceptTerms = false;

    // Constructeurs
    public RegisterDto() {}

    // Getters et Setters
    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getConfirmPassword() {
        return confirmPassword;
    }

    public void setConfirmPassword(String confirmPassword) {
        this.confirmPassword = confirmPassword;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getCompanyName() {
        return companyName;
    }

    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }

    public boolean isAcceptTerms() {
        return acceptTerms;
    }

    public void setAcceptTerms(boolean acceptTerms) {
        this.acceptTerms = acceptTerms;
    }

    /**
     * Vérifie si les mots de passe correspondent.
     * 
     * @return true si les mots de passe correspondent, false sinon
     */
    public boolean isPasswordMatching() {
        return password != null && password.equals(confirmPassword);
    }

    @Override
    public String toString() {
        return "RegisterDto{" +
                "firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", email='" + email + '\'' +
                ", phone='" + phone + '\'' +
                ", city='" + city + '\'' +
                ", country='" + country + '\'' +
                ", companyName='" + companyName + '\'' +
                ", acceptTerms=" + acceptTerms +
                '}';
    }
}

File: backend-dotnet/src/Lmp.Application/Auth/RegisterDto.cs
using System.ComponentModel.DataAnnotations;

namespace Lmp.Application.Auth;

/// <summary>Registration request body. Port of <c>com.lmp.auth.dto.RegisterDto</c>.</summary>
public sealed class RegisterDto
{
    [MaxLength(50, ErrorMessage = "Le prénom ne peut pas dépasser 50 caractères")]
    public string? FirstName { get; set; }

    [MaxLength(50, ErrorMessage = "Le nom ne peut pas dépasser 50 caractères")]
    public string? LastName { get; set; }

    [Required(ErrorMessage = "L'email est obligatoire")]
    [EmailAddress(ErrorMessage = "Format d'email invalide")]
    [MaxLength(100, ErrorMessage = "L'email ne peut pas dépasser 100 caractères")]
    public string Email { get; set; } = null!;

    [Required(ErrorMessage = "Le mot de passe est obligatoire")]
    [StringLength(100, MinimumLength = 6, ErrorMessage = "Le mot de passe doit contenir entre 6 et 100 caractères")]
    public string Password { get; set; } = null!;

    [Required(ErrorMessage = "La confirmation du mot de passe est obligatoire")]
    public string ConfirmPassword { get; set; } = null!;

    [MaxLength(20, ErrorMessage = "Le téléphone ne peut pas dépasser 20 caractères")]
    public string? Phone { get; set; }

    [MaxLength(255, ErrorMessage = "L'adresse ne peut pas dépasser 255 caractères")]
    public string? Address { get; set; }

    [MaxLength(50, ErrorMessage = "La ville ne peut pas dépasser 50 caractères")]
    public string? City { get; set; }

    [MaxLength(20, ErrorMessage = "Le code postal ne peut pas dépasser 20 caractères")]
    public string? PostalCode { get; set; }

    [MaxLength(50, ErrorMessage = "Le pays ne peut pas dépasser 50 caractères")]
    public string? Country { get; set; }

    [MaxLength(100, ErrorMessage = "Le nom de l'entreprise ne peut pas dépasser 100 caractères")]
    public string? CompanyName { get; set; }

    public bool AcceptTerms { get; set; }

    public bool IsPasswordMatching() => Password is not null && Password == ConfirmPassword;
}
